integrationTests/chainSimulator/helpers/helpers.go

```diff
@@ -0,0 +1,111 @@
+package helpers
+
+import (
+	"encoding/base64"
+	"encoding/hex"
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/data/transaction"
+	"github.com/multiversx/mx-chain-crypto-go/signing"
+	"github.com/multiversx/mx-chain-crypto-go/signing/mcl"
+	logger "github.com/multiversx/mx-chain-logger-go"
+	"github.com/stretchr/testify/require"
+)
+
+var log = logger.GetOrCreate("integrationTests/chainSimulator/helpers")
+
+func computeTxHash(chainSimulator ChainSimulator, tx *transaction.Transaction) (string, error) {
+	txBytes, err := chainSimulator.GetNodeHandler(1).GetCoreComponents().InternalMarshalizer().Marshal(tx)
+	if err != nil {
+		return "", err
+	}
+
+	txHasBytes := chainSimulator.GetNodeHandler(1).GetCoreComponents().Hasher().Compute(string(txBytes))
+	return hex.EncodeToString(txHasBytes), nil
+}
+
+// SendTxAndGenerateBlockTilTxIsExecuted will the provided transaction and generate block
+func SendTxAndGenerateBlockTilTxIsExecuted(
```

Comment:
Why not move this function, AddValidatorKeysInMultiKey and GenerateBlsPrivateKeys directly on the chainSimulator since they receive the chain simulator as parameter?

Reply:
done